octopusdeploy/schema_project_scheduled_trigger.go

```diff
@@ -0,0 +1,543 @@
+package octopusdeploy
+
+import (
+	"errors"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/actions"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/filters"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/projects"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/triggers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"time"
+)
+
+func flattenProjectScheduledTrigger(projectScheduledTrigger *triggers.ProjectTrigger) map[string]interface{} {
+	flattenedProjectScheduledTrigger := map[string]interface{}{}
+	flattenedProjectScheduledTrigger["id"] = projectScheduledTrigger.GetID()
+	flattenedProjectScheduledTrigger["name"] = projectScheduledTrigger.Name
+	flattenedProjectScheduledTrigger["description"] = projectScheduledTrigger.Description
+	flattenedProjectScheduledTrigger["project_id"] = projectScheduledTrigger.ProjectID
+	flattenedProjectScheduledTrigger["space_id"] = projectScheduledTrigger.SpaceID
+	flattenedProjectScheduledTrigger["is_disabled"] = projectScheduledTrigger.IsDisabled
+
+	actionType := projectScheduledTrigger.Action.GetActionType()
+	if actionType == actions.DeployLatestRelease {
+		deployLatestReleaseAction := projectScheduledTrigger.Action.(*actions.DeployLatestReleaseAction)
+		flattenedProjectScheduledTrigger["tenant_ids"] = deployLatestReleaseAction.Tenants
+		flattenedProjectScheduledTrigger["deploy_latest_release_action"] = []map[string]interface{}{
+			{
+				"source_environment_id":      deployLatestReleaseAction.SourceEnvironments[0],
```

Comment:
Is it expected to only take the first source environment from the list?

Reply:
Yeah, for some reason the API contract has SourceEnvironments as an array but it can only have one element in it otherwise it fails validation. Even on the frontend you can only select one environment. So rather than duplicating that logic on the provider I opted to just having a string property